src/useRouteQuery/useRouteQuery.ts

```diff
@@ -1,90 +1,111 @@
+import { LocationQuery, LocationQueryValue, RouteLocationNormalizedLoaded, Router, useRoute, useRouter } from 'vue-router'
 
-type UseRouteQuery = {
+export type UseRouteQuery = {
   query: LocationQuery,
   clear: () => void,
-  set: (key: string, value: string) => void,
+  set: (key: string, value: LocationQueryValue | LocationQueryValue[]) => void,
   get: (key: string) => LocationQueryValue | LocationQueryValue[],
   remove: (key: string) => void,
 }
 
 type QueryOperation = (query: LocationQuery) => LocationQuery
 
-export function useRouteQuery(): UseRouteQuery {
-  const route = useRoute()
-  const router = useRouter()
+function factory(): () => UseRouteQuery {
   const operations = reactive<QueryOperation[]>([])
+  let route: RouteLocationNormalizedLoaded
+  let router: Router
   let interval: ReturnType<typeof setTimeout>
 
-  const clear = (): void => {
-    router.push({ query: {} })
-  }
-
-  const set = (key: string, value: string): void => {
-    query[key] = value
-  }
-
-  const get = (key: string): LocationQueryValue | LocationQueryValue[] => {
-    return query[key]
-  }
-
-  const remove = (key: string): void => {
-    delete query[key]
-  }
-
-  const query: LocationQuery = new Proxy({}, {
-    get(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.get(target, property)
-      }
-
-      const query = applyQueryOperations(route.query, operations)
-
-      return query[property]
-    },
-    set(target, property, value) {
-      if (typeof property === 'symbol' || typeof value !== 'string') {
-        return Reflect.set(target, property, value)
-      }
-
-      const operation = updateOperationFactory(property, value)
-
-      operations.push(operation)
-
-      return true
-    },
-    deleteProperty(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.deleteProperty(target, property)
-      }
-
-      const operation = deleteOperationFactory(property)
-
-      operations.push(operation)
-
-      return true
-    },
-  })
-
   const update = (): void => {
     const query = applyQueryOperations(route.query, operations)
 
+    operations.splice(0)
+
     router.push({ query })
   }
 
   watch(operations, () => {
+    if (!operations.length) {
+      return
+    }
+
     clearInterval(interval)
     interval = setTimeout(update)
   })
 
-  return {
-    clear,
-    set,
-    get,
-    remove,
-    query,
+  const useRouteQuery = (): UseRouteQuery => {
+    router = useRouter()
+    route = useRoute()
+
+    const clear = (): void => {
+      router.push({ query: {} })
+    }
+
+    const set = (key: string, value: LocationQueryValue | LocationQueryValue[]): void => {
+      query[key] = value
+    }
+
+    const get = (key: string): LocationQueryValue | LocationQueryValue[] => {
+      return query[key]
+    }
+
+    const remove = (key: string): void => {
+      delete query[key]
+    }
+
+    const query: LocationQuery = new Proxy({}, {
```

Comment:
what's the benefit of using a proxy here? couldn't you just as easily define some non-exported methods for `get`/`set`/`delete`?

Reply:
I'm not totally following the methods piece but the idea here is that the `query` const can be used exactly the same as the `route.query` object. Anywhere you use `route.query` in code you could be able to use `query`.